{"id":"core\/exception\/conflictexception","type":"","title":"Google\\Cloud\\Core\\Exception\\ConflictException","name":"ConflictException","description":"<p>Exception thrown when a request fails due to a conflict.<\/p>\n<p>In REST context, this exception indicates a status code 409.<\/p>\n<p>Extends <a href=\"http:\/\/php.net\/manual\/en\/class.exception.php\" target=\"_blank\">Exception<\/a> &raquo; <a data-custom-type=\"cloud-core\/main\/core\/exception\/googleexception\">Google\\Cloud\\Core\\Exception\\GoogleException<\/a> &raquo; <a data-custom-type=\"cloud-core\/main\/core\/exception\/serviceexception\">Google\\Cloud\\Core\\Exception\\ServiceException<\/a><\/p>","examples":[],"resources":[],"methods":[{"id":"__construct","type":"constructor","name":"__construct","source":"Core\/src\/Exception\/ServiceException.php#L65","description":"<p>Handle previous exceptions differently here.<\/p>\n\nImplemented in <a data-custom-type=\"cloud-core\/main\/core\/exception\/serviceexception\">Google\\Cloud\\Core\\Exception\\ServiceException<\/a>","examples":[],"resources":[],"params":[{"name":"message","description":"","types":["<a href=\"http:\/\/php.net\/manual\/en\/language.types.string.php\" target=\"_blank\">string<\/a>"],"optional":false,"nullable":null},{"name":"code","description":"","types":["<a href=\"http:\/\/php.net\/manual\/en\/language.types.integer.php\" target=\"_blank\">int<\/a>"],"optional":false,"nullable":null},{"name":"serviceException","description":"","types":["<a href=\"http:\/\/php.net\/manual\/en\/class.exception.php\" target=\"_blank\">Exception<\/a>","<a href=\"http:\/\/php.net\/manual\/en\/language.types.null.php\" target=\"_blank\">null<\/a>"],"optional":false,"nullable":null},{"name":"metadata","description":"<p>Exception metadata.<\/p>","types":["<a href=\"http:\/\/php.net\/manual\/en\/language.types.array.php\" target=\"_blank\">array<\/a>"],"optional":true,"nullable":null}],"exceptions":[],"returns":[]},{"id":"hasServiceException","type":"instance","name":"hasServiceException","source":"Core\/src\/Exception\/ServiceException.php#L85","description":"<p>If $serviceException is set, return true.<\/p>\n\nImplemented in <a data-custom-type=\"cloud-core\/main\/core\/exception\/serviceexception\">Google\\Cloud\\Core\\Exception\\ServiceException<\/a>","examples":[],"resources":[],"params":[],"exceptions":[],"returns":[{"types":["<a href=\"http:\/\/php.net\/manual\/en\/language.types.boolean.php\" target=\"_blank\">bool<\/a>"],"description":""}]},{"id":"getServiceException","type":"instance","name":"getServiceException","source":"Core\/src\/Exception\/ServiceException.php#L95","description":"<p>Return the service exception object.<\/p>\n\nImplemented in <a data-custom-type=\"cloud-core\/main\/core\/exception\/serviceexception\">Google\\Cloud\\Core\\Exception\\ServiceException<\/a>","examples":[],"resources":[],"params":[],"exceptions":[],"returns":[{"types":["<a href=\"http:\/\/php.net\/manual\/en\/class.exception.php\" target=\"_blank\">Exception<\/a>","<a href=\"http:\/\/php.net\/manual\/en\/language.types.null.php\" target=\"_blank\">null<\/a>"],"description":""}]},{"id":"getMetadata","type":"instance","name":"getMetadata","source":"Core\/src\/Exception\/ServiceException.php#L103","description":"<p>Get exception metadata.<\/p>\n\nImplemented in <a data-custom-type=\"cloud-core\/main\/core\/exception\/serviceexception\">Google\\Cloud\\Core\\Exception\\ServiceException<\/a>","examples":[],"resources":[],"params":[],"exceptions":[],"returns":[]},{"id":"getErrorInfoMetadata","type":"instance","name":"getErrorInfoMetadata","source":"Core\/src\/Exception\/ServiceException.php#L113","description":"<p>Returns the metadata from the ErrorInfo part of the exception<\/p>\n\nImplemented in <a data-custom-type=\"cloud-core\/main\/core\/exception\/serviceexception\">Google\\Cloud\\Core\\Exception\\ServiceException<\/a>","examples":[],"resources":[],"params":[],"exceptions":[],"returns":[{"types":["<a href=\"http:\/\/php.net\/manual\/en\/language.types.array.php\" target=\"_blank\">array<\/a>"],"description":""}]},{"id":"getReason","type":"instance","name":"getReason","source":"Core\/src\/Exception\/ServiceException.php#L137","description":"<p>Returns the reason from the ErrorInfo part of the exception<\/p>\n\nImplemented in <a data-custom-type=\"cloud-core\/main\/core\/exception\/serviceexception\">Google\\Cloud\\Core\\Exception\\ServiceException<\/a>","examples":[],"resources":[],"params":[],"exceptions":[],"returns":[{"types":["<a href=\"http:\/\/php.net\/manual\/en\/language.types.string.php\" target=\"_blank\">string<\/a>"],"description":""}]}]}